docs: explicitly mention deviations from semver (#1689)
grafana/alloy: docs/sources/introduction/backward-compatibility.md

# {{% param "FULL_PRODUCT_NAME" %}} backward compatibility

> {{< param "FULL_PRODUCT_NAME" >}} mosty follows [semantic versioning][], with the differences documented in the [Exceptions](#exceptions) section.

Reviewer: ```suggestion
{{< param "FULL_PRODUCT_NAME" >}} mostly follows [semantic versioning][], with the differences documented in the [Exceptions](#exceptions) section.
```

> {{< param "FULL_PRODUCT_NAME" >}} mostly follows [semantic versioning][], with the differences documented in the [Exceptions](#exceptions) section.

Reviewer 2: ```suggestion
{{< param "FULL_PRODUCT_NAME" >}} follows [semantic versioning][]. [Exceptions](#exceptions) may arise that require a breaking change without a new major version.
```

I think we should avoid the vague "mostly" and just state we use semver, but exceptions can happen.

## Exceptions

> We strive to maintain backward compatibility, but there are situations that may arise that require a breaking change without a new major version, deviating from [item 8 of the semver specification]:

Reviewer 2: ```suggestion
We strive to maintain backward compatibility, but there are situations that may arise that require a breaking change without a new major version, deviating from [item 8 of the semver specification][]:
```
